src/dev/i18n/README.md

```diff
@@ -0,0 +1,175 @@
+      />
+      ...
+    </p>
+  );
+  ...
+  ```
+
+  `values` and `description` attributes are optional.\
+  JSX element can be parsed only if it is located in a JSX code block (not in a string).
+
+* **intl.formatMessage**
+
+  ```jsx
+  const MyComponentContent = ({ intl }) => (
+    <input
+      type="text"
+      placeholder={intl.formatMessage(
+        {
+          id: 'plugin_namespace.message_id',
+          defaultMessage: 'Default message string literal, {key}',
+          description: 'Message context or description'
+        },
+        {
+          key: 'value',
+        }
+      )}
+    />
+  );
+  ```
+
+  Second argument and `description` properties are optional.\
+  Callee of call expression should be either `intl.formatMessage` or `*.intl.formatMessage`.
+
+#### JavaScript (primarily server-side) (.js, .ts, .jsx, .tsx)
+
+```js
+i18n('plugin_namespace.message_id', {
+  values: {
+    key: 'value',
+  },
+  defaultMessage: 'Default message string literal, {key}',
+  description: 'Message context or description',
+});
+```
+
+or
+
+```js
+i18n.translate('plugin_namespace.message_id', {
+  values: {
+    key: 'value',
+  },
+  defaultMessage: 'Default message string literal, {key}',
+  description: 'Message context or description',
+});
+```
+
+`values` and `description` properties are optional.\
+ Expression can be parsed only if it is located in syntactically valid JS/TS code. Do not use type assertions in TypeScript for `defaultMessage` or `description` properties, id argument or the second argument of `i18n*` call expression. It is never needed for i18n engine use cases.
+
+#### Pug (.pug)
+
+```
+#{i18n('plugin_namespace.message_id', {
+  values: { key: 'value' },
+  defaultMessage: 'Default message string literal, {key}',
+  description: 'Message context or description',
+})}
+```
+
+`values` and `description` properties are optional.\
+Expression in `#{...}` is parsed as a JS expression.
+
+#### Handlebars (.handlebars, .hbs)
+
+```hbs
+{{i18n 'plugin_namespace.message_id' '{"defaultMessage": "Default message string literal", "description": "Message context or description"}'}}
+```
+
+`values` and `description` properties are optional.\
+The third token (the second argument of i18n function call) should be a string literal that contains a valid JSON.
+
+### Usage
+
+```bash
+node scripts/i18n_check --path path/to/plugin --path path/to/another/plugin --output ./translations --output-format json5
+```
+
+`path/to/plugin` is an example of path to a directory(-es) where messages searching should start. By default `--path` is `.`, it means that messages from all paths in `.i18nrc.json` will be parsed. Each specified path should start with any path in `.i18nrc.json` or be a part of it.\
+`--output` specifies a path to a directory, where `en.json` will be created, if `--output` is not provided, `en.json` generation will be skipped. It is useful if you want to validate i18n engine usage.\
```

Comment:
__nit:__ Use new line before `--output` and `--output-format`. And maybe organize it as non-numbered list?